lti_consumer/plugin/views.py

```diff
@@ -101,6 +105,45 @@ def has_block_access(user, block, course_key):
     return course_access and block_access
 
 
+def parse_custom_parameters(block):
+    """
+    Parse XBlock custom_parameters field values.
+
+    This allows us to transform the list items on the custom_parameters field into a dictionary of custom parameters
+    that can be used with the LtiConsumer1p3 set_custom_parameters method, it will also resolve any dynamic custom
+    parameter present on the field.
+
+    Args:
+        block: xblock Object to get custom_parameters from
+
+    Returns:
+        dict: Parsed custom_parameters dictionary
+
+    Raises:
+        LtiError: If custom_parameter cannot be parsed.
+    """
+    result = {}
+
+    for parameter in block.custom_parameters:
+        try:
+            parameter_key, parameter_value = [
+                string.strip()
+                for string in parameter.split('=', 1)
+            ]
+
+            # Resolve dynamic custom parameter value.
+            if re.search(CUSTOM_PARAMETER_TAG_REGEX, parameter_value):
+                parameter_value = resolve_custom_parameter_template(block, parameter_value)
+
+            result.update({parameter_key: parameter_value})
+        except ValueError as exc:
+            raise LtiError(
+                f'Invalid custom parameter format: "{parameter}". Should be "x=y" string.'
+            ) from exc
```

Comment:
Why do you need the original exception? @kuipumu

Reply:
@Squirrel18 I added it to avoid a re-raise linting error